pkg/registry/node/strategy.go

```diff
@@ -153,17 +153,30 @@ func ResourceLocation(getter ResourceGetter, connection client.ConnectionInfoGet
 	}
 	host := hostIP.String()
 
-	if portReq == "" || strconv.Itoa(ports.KubeletPort) == portReq {
-		// Ignore requested scheme, use scheme provided by GetConnectionInfo
+	// We check if we want to get a default Kubelet's transport. It happens if either:
+	// - no port is specified in request (Kubelet's port is default),
+	// - we're using Port stored as a DaemonEndpoint and requested port is a Kubelet's port stored in the DaemonEndpoint,
+	// - there's no information in the API about DaemonEnpoint (legacy cluster) and requested port is equal to ports.KubeletPort (cluster-wide config)
+	defaultKubeletPort := node.Status.DaemonEndpoints.KubeletEndpoint.Port
+	if defaultKubeletPort == 0 {
+		defaultKubeletPort = ports.KubeletPort
+	}
+	if portReq == "" || strconv.Itoa(defaultKubeletPort) == portReq {
 		scheme, port, kubeletTransport, err := connection.GetConnectionInfo(host)
```

Comment:
It does:
```
return scheme, c.Config.Port, c.Client.Transport, nil
```
Where `Config` is a KubeletConfig, and Port is populated here:
```
fs.UintVar(&s.KubeletConfig.Port, "kubelet-port", s.KubeletConfig.Port, "Kubelet port")
```

Node that we're deprecating kubelet-port flag in this PR.